neo/Network/P2P/Payloads/Transaction.cs

```diff
@@ -1,53 +1,38 @@
 using Neo.Cryptography;
 using Neo.IO;
-using Neo.IO.Caching;
 using Neo.IO.Json;
-using Neo.Ledger;
 using Neo.Persistence;
 using Neo.SmartContract;
+using Neo.SmartContract.Native;
 using Neo.VM;
+using Neo.Wallets;
 using System;
 using System.Collections.Generic;
 using System.IO;
 using System.Linq;
-using System.Text;
 
 namespace Neo.Network.P2P.Payloads
 {
-    public abstract class Transaction : IEquatable<Transaction>, IInventory
+    public sealed class Transaction : IEquatable<Transaction>, IInventory
     {
         public const int MaxTransactionSize = 102400;
         /// <summary>
         /// Maximum number of attributes that can be contained within a transaction
         /// </summary>
-        private const int MaxTransactionAttributes = 16;
+        private const int MaxCosigners = 16;
 
-        /// <summary>
-        /// Reflection cache for TransactionType
-        /// </summary>
-        private static ReflectionCache<byte> ReflectionCache = ReflectionCache<byte>.CreateFromEnum<TransactionType>();
-
-        public readonly TransactionType Type;
         public byte Version;
-        public TransactionAttribute[] Attributes;
-        public CoinReference[] Inputs;
-        public TransactionOutput[] Outputs;
+        public byte[] Script;
+        public long Gas;
+        public long NetworkFee;
+        public UInt160[] Cosigners;
```

Comment:
Could you explain me what is the `Cosigners`? Are the contracts for which our signatures will be valid?

Reply:
It is the same as `TransactionAttributeUsage.Script`.